src/node/desktop/test/unit/main/desktop-options.test.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * desktop-options.test.ts
+ * 
+ * Copyright (C) 2021 by RStudio, PBC
+ * 
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ */
+
+import { describe } from 'mocha';
+import { assert } from 'chai';
+
+import { TestDesktopOptions, DesktopOptionsImpl, kDesktopOptionDefaults, clearOptionsSingleton } from '../../../src/main/desktop-options';
+import { FilePath } from '../../../src/core/file-path';
+import { Err, isSuccessful } from '../../../src/core/err';
+
+const kTestingConfigFilename = process.cwd() + 'DesktopOptionsTesting';
```

Comment:
We should write any temporary files created during tests somewhere in the temp path, and not use a fully hardcoded name (so any tests run in parallel don't fight each other; just a theoretical problem for now). I do stuff like that in the file-path tests, look for `os.tmpdir()` and `randomString()`. Maybe create a help for that in `unit-utils.ts`.

Reply:
Added some to create either random directories or files, let me know if they're close to what you had in mind